### ISSUE TITLE
Codist Beta 7.8

### ISSUE THREAD

**Issue by wmjordan** [wmjordan/Codist#311]
## Download
[Codist 9554](https://github.com/wmjordan/Codist/files/13958244/Codist.zip)

[Codist 9550](https://github.com/wmjordan/Codist/files/13931412/Codist.zip)
[Codist 9546](https://github.com/wmjordan/Codist/files/13873215/Codist.zip)
[Codist 9530](https://github.com/wmjordan/Codist/files/13802617/Codist.zip)
[Codist 9521](https://github.com/wmjordan/Codist/files/13797744/Codist.zip)
[Codist 9510](https://github.com/wmjordan/Codist/files/13774809/Codist.zip)
[Codist 9496](https://github.com/wmjordan/Codist/files/13763256/Codist.zip)
[Codist 9482](https://github.com/wmjordan/Codist/files/13757806/Codist.zip)
[Codist 9465](https://github.com/wmjordan/Codist/files/13741912/Codist.zip)

## What's new
- General
   - [x] Prevented CSharpSymbolContextMenu commands from crashing VS
   - [x] Fixed potential crash when getting source from implicit constructor
- Syntax Highlight
   - [x] Made Comment Tagger compatible with the new version of VS
   - [x] Addressed some weird problems of syntax highlight (#303 , #309 , etc. )
   - [x] Reenabled C# Syntax Highlight in preview text view
   - [x] Fixed potential VS deadlock caused by Comment Tagger
- Super Quick Info
   - [x] Fixed a potential crash when advanced Syntax Highlight, NaviBar, etc. features were disabled (#307)
   - [x] Improved the display of some C# XML Doc (#306)
- NaviBar
   - [x] Added filter buttons to Markdown Navi Bar to filter to specific heading levels
        <img width="218" alt="image" src="https://github.com/wmjordan/Codist/assets/12182730/fcce0195-4de0-4c5b-b14b-b17f04d76381">
- Smart Bar
   - [x] Added Sort Lines, Join Lines to Smart Bar
   - [x] Added Delete Statement, Delete Field and Delete Event refactoring
   - [x] Fixed SmartBar would not show up in some cases

**Comment by wmjordan** [comment]
@fitdev, @laicasaane, @nurbles
Hey, my friends, I am happy to tell you that there is a new beta, which is supposed to improve the stability of the Advanced Syntax Highlight feature. Please download and give it a try.

**Comment by fitdev** [comment]
Thank you for the new beta! I will let you know if it fixes the issue with comment highlights disappearing once I had a chance to test it.

But there is one long standing issue, which perhaps can be addressed: improper styling of type names when inheritance / interfaces are involved. (There is also an additional issue with ref/readonly/record structs, but it is too complicated I take it).

![Screenshot_20231223_133825](https://github.com/wmjordan/Codist/assets/8166847/1f25da06-36b9-4008-bd65-cb5d9042c093)

![Screenshot_20231223_133739](https://github.com/wmjordan/Codist/assets/8166847/e39b1cba-dd6a-42da-b0ca-84a4dac02d6e)

![Screenshot_20231223_133757](https://github.com/wmjordan/Codist/assets/8166847/42208bc4-8b01-4f0c-8d97-dfd0e28fac96)

For some reason they are getting classified as "Abstract Members" even though they are types. This causes the color to be wrong. I this really a bug / issue, or is my configuration wrong?

**Comment by wmjordan** [comment]
Oh, it is historical mistake of naming. Actually it should be "Abstract symbol".
Despite of the name, how do you think whether abstract types should also be highlighted with a special "abstract" style?
You may try to use gradient background color to denote abstract symbols, so the style of abstraction and type kind can coexist.

**Comment by fitdev** [comment]
Oh I see. Thank you for the explanation. Yeah, I should experiment with the background perhaps.
Though in my case I am mainly just interested in seeing which _members_ are _virtual/abstract_ and would like all other abstract symbols' styles to remain unchanged. Wish Roslyn would offer finer API so that Codist could distinguish such cases.

**Comment by wmjordan** [comment]
I found my self did not use the "Abstract Member" style--perhaps it is the reason that you'd pointed out above.
I will limit the Abstract Member style to type members only, not on types, in the next version.

**Comment by wmjordan** [comment]
@nurbles
I just reported the ["identifer" issue](https://developercommunity.visualstudio.com/t/Identifier-has-too-high-priority-in-sy/10547451?port=1026&fsid=10694cc2-b671-40cb-a5e9-ed7ebd3004aa&ref=native&refTime=1703422154034&refUserId=f9708ee9-135e-4641-8d26-c0a7831ead32) that [you told me several months ago](https://github.com/wmjordan/Codist/issues/282). It was the problem of VS, not Codist. I disabled Codist and still got weird results.

**Comment by fitdev** [comment]
After testing the new version for a day. It appears there are no more user comments custom tag highlighting issues - works great as before! Thank you for this awesome work!

**Comment by wmjordan** [comment]
@fitdev 
Good.
Please test the new beta, which limits the "Abstract Member" style to type members only.

**Comment by fitdev** [comment]
That was so fast! Thank you! Works great!

**Comment by kwonganding** [comment]
very 不错！

**Comment by kwonganding** [comment]
@wmjordan XAML文件的语法高亮没了
Syntax highlighting is gone for XAML files

**Comment by wmjordan** [comment]
> @wmjordan XAML文件的语法高亮没了 Syntax highlighting is gone for XAML files

是哦。请还原回旧版本。

**Comment by kwonganding** [comment]
--依然存在VS卡死的问题，之前的版本也是有的，而且几率更高。
* 使用VS过程中，偶发（高频率出现）VS卡死情况，就常规的Codlist操作，如鼠标悬停代码提示、按住Ctrl键+点击转到定义（这个出现几率比较高）、工具条，智能导航使用。
* 环境说明：，项目A，自建解决方案，不到20个项目，代码量不多，使用没有遇到过问题。方案B，200+个项目，有很多大文件，10000+的代码文件很多，VS卡死几率很高，禁用Codelist后正常。

**Comment by wmjordan** [comment]
卡死时，请用 [ProcDump](https://learn.microsoft.com/en-us/sysinternals/downloads/procdump) 做一个迷你内存转储。发上来让我看看是在哪里卡住了。

**Comment by kwonganding** [comment]
@wmjordan  如下，超过25MB的限制，分成三个包
[devenv.exe_231226_112430.zip](https://github.com/wmjordan/Codist/files/13768326/devenv.exe_231226_112430.zip)
[devenv.exe_231226_112430-2.zip](https://github.com/wmjordan/Codist/files/13768570/devenv.exe_231226_112430-2.zip)
[devenv.exe_231226_112430-1.zip](https://github.com/wmjordan/Codist/files/13768576/devenv.exe_231226_112430-1.zip)

**Comment by kwonganding** [comment]
@wmjordan 用VS附件看了下，卡主的地方可能是下图：
![image](https://github.com/wmjordan/Codist/assets/16829626/519b0926-d037-4c03-98d6-18da6edd001d)

**Comment by wmjordan** [comment]
> @wmjordan 用VS附件看了下，卡主的地方可能是下图： ![image](https://private-user-images.githubusercontent.com/16829626/292811867-519b0926-d037-4c03-98d6-18da6edd001d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM1NzU0OTEsIm5iZiI6MTcwMzU3NTE5MSwicGF0aCI6Ii8xNjgyOTYyNi8yOTI4MTE4NjctNTE5YjA5MjYtZDAzNy00YzAzLTk4ZDYtMThkYTZlZGQwMDFkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI2VDA3MTk1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI3OGM1Zjg5ZjA5Yjc5NmQxZjk2OWNhNzgyM2M3N2Y3ZGY3MmQ3NmQ0NWZkMDA2YjIyZjczZWEzOThmMjlmYzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.cDFMdh5oO6gaEx0_6KM3CSjC4fJdY6-EqSvfxaGhT4g)

太诡异了。那个地方不应该陷入死循环的。
程序的意图是遍历包含一个成员的外部类型，如果外层类型不存在，应该返回 null，跳出循环。
<img width="563" alt="image" src="https://github.com/wmjordan/Codist/assets/12182730/4bc0673b-946a-4f62-be24-6907ed0c3026">

请下载新的测试版。新测试版会在遇到10层嵌套类后弹出一个对话框。请看看是否会弹出这样的对话框，如果有，请尝试构造一个能重现问题的解决方案发上来。

**Comment by kwonganding** [comment]
@wmjordan 
使用最新版依然有多次卡死，但并没有弹出对话框。代码不方便发，尝试构建重现代码解决方案但不成功，后面有时间了我在试试。
1、卡死前的操作基本都是在智能提示上，鼠标放到某一个代码上。
2、卡死时应该不是死循环，半个小时后恢复正常，好多次我没管他后面会恢复。

**可能是我这项目的情况比较特殊**：
解决方案中项目有200+，很多代码比较长。这个应该不是重点，重点是我不是加载的解决方案，是加载“文件夹”的方式，解决方案下全是文件夹、代码。没有解决方案文件，项目之间的引用是不完整的，很多代码也是不完整的，就是说整个解决方案中的代码是比较分散、杂乱的，就是一堆代码，。
1、所以在做类型关系查找时很高概率出问题，可能是VS本身也有相关Bug。我另一个正常解决方案是没出过问题的。
2、可能只做一些健壮性改进，如超时处理。
![image](https://github.com/wmjordan/Codist/assets/16829626/6e2ff5be-a054-462b-8e88-0b3df3ccf4c6)

**Comment by wmjordan** [comment]
根据你描述的场景，我还是无法重现故障。我尝试使用VS打开一个有二百多个项目、一万多个代码文件的文件夹，然后尝试打开里面的一些代码文件，显示代码提示时没有遇到卡死的情况。
按照你提供的转储文件和上面的截图，程序的确是停在了`ShowContainingTypes`方法里，上面上传的9510测试版应该会检测到里面的异常循环数量——毕竟那个循环体非常简单，只有三行代码。新的测试版在添加项到`n`这个变量后会检查已添加了多少项。如果不断地添加项，就会弹出提示框。
方法中除了那个循环之外，也没其它可能导致卡死的代码了。

**Comment by fitdev** [comment]
The new beta version was very stable with user comments custom tag highlighting - lasting for a few days. Then all of a sudden, as I was typing text, user comment highlights were gone - _everywhere_ - in all opened documents. Closing and reopening documents _did NOT_ restore them. So I had to restart VS to restore highlights.

**Comment by fitdev** [comment]
Also just noticed that syntax highlighting for xaml, axaml, csproj (all xml-based) files is gone. Interestingly user-tagged comments in there do get highlighted however.

**Comment by wmjordan** [comment]
> Also just noticed that syntax highlighting for xaml, axaml, csproj (all xml-based) files is gone.

Yes. It was reported above by another Chinese user.
I am looking into it.

**Comment by wmjordan** [comment]
@kwonganding @fitdev 
The new beta should properly restore the default XAML and XML highlight styles.

**Comment by fitdev** [comment]
Thank you for such a quick fix! Works really well, except that XML user-tagged comments do not get their foreground color properly applied:

![Screenshot_20231230_133411](https://github.com/wmjordan/Codist/assets/8166847/81750c35-f6bc-483c-afd0-76f5200d8b74)
![Screenshot_20231230_133420](https://github.com/wmjordan/Codist/assets/8166847/05e013aa-568a-472d-aff6-a512144cfd5e)

**Comment by wmjordan** [comment]
Thank you for your feedback.
There are still bugs. But I feel that I am almost getting the ultimate algorithm of sorting highlight styles.

**Comment by wmjordan** [comment]
Please download the new beta, which better addressed the syntax highlight priority issue.

**Comment by laicasaane** [comment]
I've been experiencing this issue for a while, not sure since 7.6 or 7.7. Whenever I have Navigation Bar enabled, VS will be slowing down for code typing.

**Comment by fitdev** [comment]
The latest beta, too, after quite some time working in VS looses all custom comment tag highlighting in all file types. restoring highlights in XML-based files requires simply typing in the custom comment tag prefix there. But for `*.cs` files, the only fix is to restart VS. Closing and reopening the document or typing does not restore comment custom tag highlights. This however seems to happen only after using VS for a while (at least a few hours). Otherwise it is pretty stable.

**Comment by wmjordan** [comment]
Thank you for your comments.
I will take a look at the comment tagger, when I have time.
We are getting to be busy recently and do not expect it will be fast.

**Comment by wmjordan** [comment]
@fitdev 
Please download the new beta.
Hopefully this can work well.

**Comment by wmjordan** [comment]
> I've been experiencing this issue for a while, not sure since 7.6 or 7.7. Whenever I have Navigation Bar enabled, VS will be slowing down for code typing.

I looked into the code and suspected that the awaited async task was running in the UI thread, which once occurred with in the C# syntax highlighting module in Codist and eventually I solved it by explicitly creating a new thread to run time consuming procedures. Are you experienced in dealing with `Task`s? Please help take a look into the [Update method](https://github.com/wmjordan/Codist/blob/37528a628d03136b5938671646489fdf82be4d03/Codist/NaviBar/CSharpBar.cs#L137) which might be the cause of freezing your VS.

**Comment by laicasaane** [comment]
Unfortunately, I'm more experienced in game coding 😂

**Comment by fitdev** [comment]
Have been testing the latest beta 9546 for a few days. So far everything worked smoothly and comment tags did not disappear.
However VS became really unresponsive and almost froze due to Codist's comment tagger consuming all the resources of a UI thread:

```
0, ntdll.dll!NtWaitForSingleObject+0x14
1, KernelBase.dll!WaitForSingleObjectEx+0x8e
2, clr.dll!StrongNameTokenFromPublicKey+0x142ea
3, clr.dll!StrongNameTokenFromPublicKey+0x1428b
4, clr.dll!StrongNameTokenFromPublicKey+0x141f3
5, clr.dll!TranslateSecurityAttributes+0x139d05
6, clr.dll!DllCanUnloadNowInternal+0xf6a7
7, clr.dll!DllCanUnloadNowInternal+0x11306
8, clr.dll!DllCanUnloadNowInternal+0x10383
9, clr.dll!IsOS+0x28d4
10, clr.dll!CreateApplicationContext+0x134b1
11, clr.dll!CoUninitializeEE+0x2ecc0
12, System.Collections.Generic.Dictionary`2[[System.__Canon, mscorlib],[System.Boolean, mscorlib]].Resize(Int32, Boolean) + 0x2c <-- mscorlib.ni.dll+0x52070c
13, System.Collections.Generic.Dictionary`2[[System.__Canon, mscorlib],[System.Boolean, mscorlib]].Insert(System.__Canon, Boolean, Boolean) + 0x221 <-- mscorlib.ni.dll+0x5205e1
14, Codist.Taggers.CommentTagger.IsComment(Microsoft.VisualStudio.Text.Classification.IClassificationType) + 0xa7 <-- 0x7ffc57d9fb57
15, Codist.Taggers.CommentTagger.TagComments(Microsoft.VisualStudio.Text.SnapshotSpan, Microsoft.VisualStudio.Text.Tagging.IMappingTagSpan`1<Microsoft.VisualStudio.Text.Tagging.IClassificationTag>) + 0x68 <-- 0x7ffc57d9f5f8
16, Codist.Taggers.CommentTagger+CSharpCommentTagger.TagComments(Microsoft.VisualStudio.Text.SnapshotSpan, Microsoft.VisualStudio.Text.Tagging.IMappingTagSpan`1<Microsoft.VisualStudio.Text.Tagging.IClassificationTag>) + 0x1d1 <-- 0x7ffc57d9f561
17, Codist.Taggers.CommentTagger+<GetTags>d__22.MoveNext() + 0x1a9 <-- 0x7ffc57d9e829
18, Microsoft.VisualStudio.Text.Tagging.Implementation.TagAggregator`1+<GetTagsForBuffer>d__49[[System.__Canon, mscorlib]].MoveNext() + 0x518 <-- Microsoft.VisualStudio.Platform.VSEditor.ni.dll+0x723ec8
19, Microsoft.VisualStudio.Text.Classification.Implementation.ClassifierAggregator.InternalGetClassificationSpans(Microsoft.VisualStudio.Text.SnapshotSpan, System.Collections.Generic.IEnumerable`1<Microsoft.VisualStudio.Text.Tagging.IMappingTagSpan`1<Microsoft.VisualStudio.Text.Tagging.IClassificationTag>>) + 0x63 <-- Microsoft.VisualStudio.Platform.VSEditor.ni.dll+0x72ac73
20, Microsoft.VisualStudio.Text.Classification.Implementation.ClassifierAggregator.GetClassificationSpans(Microsoft.VisualStudio.Text.SnapshotSpan) + 0x75 <-- Microsoft.VisualStudio.Platform.VSEditor.ni.dll+0x72ab15
21, Microsoft.VisualStudio.Text.Formatting.Implementation.NormalizedSpanGenerator.CreateClassifiedRuns(Microsoft.VisualStudio.Text.SnapshotSpan, Microsoft.VisualStudio.Text.Classification.IAccurateClassifier, Microsoft.VisualStudio.Text.Classification.IClassificationFormatMap, Int32 ByRef, Int32, System.Nullable`1<System.Threading.CancellationToken>) + 0x84 <-- Microsoft.VisualStudio.Platform.VSEditor.ni.dll+0x72a314
22, Microsoft.VisualStudio.Text.Formatting.Implementation.NormalizedSpanGenerator.Create(Microsoft.VisualStudio.Text.Formatting.ITextAndAdornmentCollection, Microsoft.VisualStudio.Text.ITextSnapshot, Microsoft.VisualStudio.Text.Classification.IAccurateClassifier, Microsoft.VisualStudio.Text.Classification.IClassificationFormatMap, Int32, System.Nullable`1<System.Threading.CancellationToken>) + 0x18c <-- Microsoft.VisualStudio.Platform.VSEditor.ni.dll+0x729eec
23, Microsoft.VisualStudio.Text.Formatting.Implementation.FormattedLineSource.FormatLineInVisualBufferIfChanged(Microsoft.VisualStudio.Text.ITextSnapshotLine, System.Collections.Generic.IList`1<Microsoft.VisualStudio.Text.Formatting.IFormattedLine>, System.Nullable`1<System.Threading.CancellationToken>) + 0x81 <-- Microsoft.VisualStudio.Platform.VSEditor.ni.dll+0x796cb1
24, Microsoft.VisualStudio.Text.Editor.Implementation.WpfTextView.FormatSnapshotLine(Microsoft.VisualStudio.Text.ITextSnapshotLine, System.Nullable`1<System.Threading.CancellationToken>) + 0x1f1 <-- Microsoft.VisualStudio.Platform.VSEditor.ni.dll+0x728521
25, Microsoft.VisualStudio.Text.Editor.Implementation.WpfTextView.LayoutRemainingLines(LayoutDescription, System.Nullable`1<System.Threading.CancellationToken>, Boolean, Boolean, Boolean) + 0x16c <-- Microsoft.VisualStudio.Platform.VSEditor.ni.dll+0x7311ec
26, Microsoft.VisualStudio.Text.Editor.Implementation.WpfTextView.PerformLayout(Microsoft.VisualStudio.Text.SnapshotPoint, Double, Microsoft.VisualStudio.Text.Editor.ViewRelativePosition, Double, Double, Boolean, Boolean, System.Nullable`1<System.Threading.CancellationToken>) + 0x433 <-- Microsoft.VisualStudio.Platform.VSEditor.ni.dll+0x720d83
27, Microsoft.VisualStudio.Text.Editor.Implementation.WpfTextView.PerformLayout() + 0xdf <-- Microsoft.VisualStudio.Platform.VSEditor.ni.dll+0x78319f
28, Microsoft.VisualStudio.Text.Editor.Implementation.WpfTextView.<QueueLayout>b__283_1() + 0x34 <-- Microsoft.VisualStudio.Platform.VSEditor.ni.dll+0x772c84
29, Microsoft.VisualStudio.Text.Editor.Implementation.WpfTextView.<QueueLayout>b__283_0() + 0x5b <-- Microsoft.VisualStudio.Platform.VSEditor.ni.dll+0x7470ab
30, System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate, System.Object, Int32) + 0xaa <-- WindowsBase.ni.dll+0x142d3a
31, System.Windows.Threading.ExceptionWrapper.TryCatchWhen(System.Object, System.Delegate, System.Object, Int32, System.Delegate) + 0x36 <-- WindowsBase.ni.dll+0x142bf6
32, System.Windows.Threading.DispatcherOperation.InvokeImpl() + 0x10c <-- WindowsBase.ni.dll+0x1456dc
33, MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(System.Object) + 0x4d <-- WindowsBase.ni.dll+0x15926d
34, System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean) + 0x108 <-- mscorlib.ni.dll+0x54e9a8
35, System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean) + 0x15 <-- mscorlib.ni.dll+0x54e895
36, System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object) + 0x55 <-- mscorlib.ni.dll+0x54e865
37, MS.Internal.CulturePreservingExecutionContext.Run(MS.Internal.CulturePreservingExecutionContext, System.Threading.ContextCallback, System.Object) + 0xb1 <-- WindowsBase.ni.dll+0x1454d1
38, System.Windows.Threading.DispatcherOperation.Invoke() + 0x62 <-- WindowsBase.ni.dll+0x1452d2
39, System.Windows.Threading.Dispatcher.ProcessQueue() + 0x1fd <-- WindowsBase.ni.dll+0x14086d
40, System.Windows.Threading.Dispatcher.WndProcHook(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef) + 0x6f <-- WindowsBase.ni.dll+0x13f21f
41, MS.Win32.HwndWrapper.WndProc(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef) + 0xc4 <-- WindowsBase.ni.dll+0x1422b4
42, MS.Win32.HwndSubclass.DispatcherCallbackOperation(System.Object) + 0x84 <-- WindowsBase.ni.dll+0x1426b4
43, System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate, System.Object, Int32) + 0x68 <-- WindowsBase.ni.dll+0x142cf8
44, System.Windows.Threading.ExceptionWrapper.TryCatchWhen(System.Object, System.Delegate, System.Object, Int32, System.Delegate) + 0x36 <-- WindowsBase.ni.dll+0x142bf6
45, System.Windows.Threading.Dispatcher.LegacyInvokeImpl(System.Windows.Threading.DispatcherPriority, System.TimeSpan, System.Delegate, System.Object, Int32) + 0x172 <-- WindowsBase.ni.dll+0x140202
46, MS.Win32.HwndSubclass.SubclassWndProc(IntPtr, Int32, IntPtr, IntPtr) + 0x152 <-- WindowsBase.ni.dll+0x142022
47, DomainNeutralILStubClass.IL_STUB_ReversePInvoke(Int64, Int32, Int64, Int64) + 0x52 <-- WindowsBase.ni.dll+0x3224d2
48, clr.dll!DllCanUnloadNowInternal+0x419e
49, user32.dll!DispatchMessageW+0x741
50, user32.dll!DispatchMessageW+0x201
51, msenv.dll!DllGetClassObject+0x18b0b
52, msenv.dll!DllGetClassObject+0x18ee8
53, msenv.dll!DllGetClassObject+0x91d33
54, msenv.dll!DllGetClassObject+0x92041
55, msenv.dll!DllGetClassObject+0x91f0f
56, msenv.dll!DllGetClassObject+0x91ea9
57, msenv.dll!DllGetClassObject+0x91bed
58, msenv.dll!VStudioMain+0xdc
59, devenv.exe+0xd594
60, devenv.exe+0x862f
61, devenv.exe+0x30c8
62, devenv.exe!SetOnAssertCallback+0x416f2
63, kernel32.dll!BaseThreadInitThunk+0x1d
64, ntdll.dll!RtlUserThreadStart+0x28
```

Closing the document did not help, so had to restart VS. I suppose it could be a one-time thing though. Because otherwise everything worked perfectly! I will see if it repeats and let you know.

Another issue I noticed, which I do not know if it is Codist-related, but it seems to have started after one of the last recent beta upgrades, is that: when I select text with a mouse, _sometimes_ it would stop the selection operation and begin drag operation or just stop the selection without starting the drag operation (even though I am still selecting text and have not released the left mouse button).

**Comment by wmjordan** [comment]
@fitdev 
Please download the new beta which addresses the infinite loop issue.

> _sometimes_ it would stop the selection operation and begin drag operation or just stop the selection without starting the drag operation

This never happens to me. Codist does not change your selection until you explicitly execute some commands from the NaviBar or SmartBar. Your mouse button might be broken.

**Comment by fitdev** [comment]
Strangely now it seems much worse. It freezes for the very start:

```
0, clr.dll!GetMetaDataPublicInterfaceFromInternal+0xa379
1, System.Collections.Generic.ObjectEqualityComparer`1[[System.__Canon, mscorlib]].GetHashCode(System.__Canon) + 0x17 <-- mscorlib.ni.dll+0x51a417
2, System.Collections.Generic.Dictionary`2[[System.__Canon, mscorlib],[System.Boolean, mscorlib]].Insert(System.__Canon, Boolean, Boolean) + 0x67 <-- mscorlib.ni.dll+0x520427
3, System.Collections.Generic.Dictionary`2[[System.__Canon, mscorlib],[System.Boolean, mscorlib]]..ctor(System.Collections.Generic.IDictionary`2<System.__Canon,Boolean>, System.Collections.Generic.IEqualityComparer`1<System.__Canon>) + 0x10a <-- mscorlib.ni.dll+0xd8e5ca
4, Codist.Taggers.CommentTagger.IsComment(Microsoft.VisualStudio.Text.Classification.IClassificationType) + 0x6a <-- 0x7ffc57da4dea
5, Codist.Taggers.CommentTagger.TagComments(Microsoft.VisualStudio.Text.SnapshotSpan, Microsoft.VisualStudio.Text.Tagging.IMappingTagSpan`1<Microsoft.VisualStudio.Text.Tagging.IClassificationTag>) + 0x68 <-- 0x7ffc57da48c8
6, Codist.Taggers.CommentTagger+CSharpCommentTagger.TagComments(Microsoft.VisualStudio.Text.SnapshotSpan, Microsoft.VisualStudio.Text.Tagging.IMappingTagSpan`1<Microsoft.VisualStudio.Text.Tagging.IClassificationTag>) + 0x1d1 <-- 0x7ffc57da4831
7, Codist.Taggers.CommentTagger+<GetTags>d__22.MoveNext() + 0x1a9 <-- 0x7ffc57da3559
8, Microsoft.VisualStudio.Text.Tagging.Implementation.TagAggregator`1+<GetTagsForBuffer>d__49[[System.__Canon, mscorlib]].MoveNext() + 0x518 <-- Microsoft.VisualStudio.Platform.VSEditor.ni.dll+0x723ec8
9, Microsoft.VisualStudio.Text.Classification.Implementation.ClassifierAggregator.InternalGetClassificationSpans(Microsoft.VisualStudio.Text.SnapshotSpan, System.Collections.Generic.IEnumerable`1<Microsoft.VisualStudio.Text.Tagging.IMappingTagSpan`1<Microsoft.VisualStudio.Text.Tagging.IClassificationTag>>) + 0x63 <-- Microsoft.VisualStudio.Platform.VSEditor.ni.dll+0x72ac73
10, Microsoft.VisualStudio.Text.Classification.Implementation.ClassifierAggregator.GetClassificationSpans(Microsoft.VisualStudio.Text.SnapshotSpan) + 0x75 <-- Microsoft.VisualStudio.Platform.VSEditor.ni.dll+0x72ab15
11, Microsoft.VisualStudio.Text.Formatting.Implementation.NormalizedSpanGenerator.CreateClassifiedRuns(Microsoft.VisualStudio.Text.SnapshotSpan, Microsoft.VisualStudio.Text.Classification.IAccurateClassifier, Microsoft.VisualStudio.Text.Classification.IClassificationFormatMap, Int32 ByRef, Int32, System.Nullable`1<System.Threading.CancellationToken>) + 0x84 <-- Microsoft.VisualStudio.Platform.VSEditor.ni.dll+0x72a314
12, Microsoft.VisualStudio.Text.Formatting.Implementation.NormalizedSpanGenerator.Create(Microsoft.VisualStudio.Text.Formatting.ITextAndAdornmentCollection, Microsoft.VisualStudio.Text.ITextSnapshot, Microsoft.VisualStudio.Text.Classification.IAccurateClassifier, Microsoft.VisualStudio.Text.Classification.IClassificationFormatMap, Int32, System.Nullable`1<System.Threading.CancellationToken>) + 0x18c <-- Microsoft.VisualStudio.Platform.VSEditor.ni.dll+0x729eec
13, Microsoft.VisualStudio.Text.Formatting.Implementation.FormattedLineSource.FormatLineInVisualBufferIfChanged(Microsoft.VisualStudio.Text.ITextSnapshotLine, System.Collections.Generic.IList`1<Microsoft.VisualStudio.Text.Formatting.IFormattedLine>, System.Nullable`1<System.Threading.CancellationToken>) + 0x81 <-- Microsoft.VisualStudio.Platform.VSEditor.ni.dll+0x796cb1
14, Microsoft.VisualStudio.Text.Editor.Implementation.WpfTextView.FormatSnapshotLine(Microsoft.VisualStudio.Text.ITextSnapshotLine, System.Nullable`1<System.Threading.CancellationToken>) + 0x1f1 <-- Microsoft.VisualStudio.Platform.VSEditor.ni.dll+0x728521
15, Microsoft.VisualStudio.Text.Editor.Implementation.WpfTextView.LayoutRemainingLines(LayoutDescription, System.Nullable`1<System.Threading.CancellationToken>, Boolean, Boolean, Boolean) + 0x16c <-- Microsoft.VisualStudio.Platform.VSEditor.ni.dll+0x7311ec
16, Microsoft.VisualStudio.Text.Editor.Implementation.WpfTextView.PerformLayout(Microsoft.VisualStudio.Text.SnapshotPoint, Double, Microsoft.VisualStudio.Text.Editor.ViewRelativePosition, Double, Double, Boolean, Boolean, System.Nullable`1<System.Threading.CancellationToken>) + 0x433 <-- Microsoft.VisualStudio.Platform.VSEditor.ni.dll+0x720d83
17, Microsoft.VisualStudio.Text.Editor.Implementation.WpfTextView.PerformLayout() + 0xdf <-- Microsoft.VisualStudio.Platform.VSEditor.ni.dll+0x78319f
18, Microsoft.VisualStudio.Text.Editor.Implementation.WpfTextView.<QueueLayout>b__283_1() + 0x34 <-- Microsoft.VisualStudio.Platform.VSEditor.ni.dll+0x772c84
19, Microsoft.VisualStudio.Text.Editor.Implementation.WpfTextView.<QueueLayout>b__283_0() + 0x5b <-- Microsoft.VisualStudio.Platform.VSEditor.ni.dll+0x7470ab
20, System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate, System.Object, Int32) + 0xaa <-- WindowsBase.ni.dll+0x142d3a
21, System.Windows.Threading.ExceptionWrapper.TryCatchWhen(System.Object, System.Delegate, System.Object, Int32, System.Delegate) + 0x36 <-- WindowsBase.ni.dll+0x142bf6
22, System.Windows.Threading.DispatcherOperation.InvokeImpl() + 0x10c <-- WindowsBase.ni.dll+0x1456dc
23, MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(System.Object) + 0x4d <-- WindowsBase.ni.dll+0x15926d
24, System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean) + 0x108 <-- mscorlib.ni.dll+0x54e9a8
25, System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean) + 0x15 <-- mscorlib.ni.dll+0x54e895
26, System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object) + 0x55 <-- mscorlib.ni.dll+0x54e865
27, MS.Internal.CulturePreservingExecutionContext.Run(MS.Internal.CulturePreservingExecutionContext, System.Threading.ContextCallback, System.Object) + 0xb1 <-- WindowsBase.ni.dll+0x1454d1
28, System.Windows.Threading.DispatcherOperation.Invoke() + 0x62 <-- WindowsBase.ni.dll+0x1452d2
29, System.Windows.Threading.Dispatcher.ProcessQueue() + 0x1fd <-- WindowsBase.ni.dll+0x14086d
30, System.Windows.Threading.Dispatcher.WndProcHook(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef) + 0x6f <-- WindowsBase.ni.dll+0x13f21f
31, MS.Win32.HwndWrapper.WndProc(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef) + 0xc4 <-- WindowsBase.ni.dll+0x1422b4
32, MS.Win32.HwndSubclass.DispatcherCallbackOperation(System.Object) + 0x84 <-- WindowsBase.ni.dll+0x1426b4
33, System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate, System.Object, Int32) + 0x68 <-- WindowsBase.ni.dll+0x142cf8
34, System.Windows.Threading.ExceptionWrapper.TryCatchWhen(System.Object, System.Delegate, System.Object, Int32, System.Delegate) + 0x36 <-- WindowsBase.ni.dll+0x142bf6
35, System.Windows.Threading.Dispatcher.LegacyInvokeImpl(System.Windows.Threading.DispatcherPriority, System.TimeSpan, System.Delegate, System.Object, Int32) + 0x172 <-- WindowsBase.ni.dll+0x140202
36, MS.Win32.HwndSubclass.SubclassWndProc(IntPtr, Int32, IntPtr, IntPtr) + 0x152 <-- WindowsBase.ni.dll+0x142022
37, DomainNeutralILStubClass.IL_STUB_ReversePInvoke(Int64, Int32, Int64, Int64) + 0x52 <-- WindowsBase.ni.dll+0x3224d2
38, clr.dll!DllCanUnloadNowInternal+0x419e
39, user32.dll!DispatchMessageW+0x741
40, user32.dll!DispatchMessageW+0x201
41, msenv.dll!DllGetClassObject+0x18b0b
42, msenv.dll!DllGetClassObject+0x18ee8
43, msenv.dll!DllGetClassObject+0x91d33
44, msenv.dll!DllGetClassObject+0x92041
45, msenv.dll!DllGetClassObject+0x91f0f
46, msenv.dll!DllGetClassObject+0x91ea9
47, msenv.dll!DllGetClassObject+0x91bed
48, msenv.dll!VStudioMain+0xdc
49, devenv.exe+0xd594
50, devenv.exe+0x862f
51, devenv.exe+0x30c8
52, devenv.exe!SetOnAssertCallback+0x416f2
53, kernel32.dll!BaseThreadInitThunk+0x1d
54, ntdll.dll!RtlUserThreadStart+0x28
```

Going back to 9546.

> Your mouse button might be broken.

I thought so at first, but it happens with different mice and only in VS. Perhaps it's just an issue with VS itself which they will fix or something weird on my end. I will let you know if I discover something related to Codist with regards to this issue.

**Comment by wmjordan** [comment]
Sorry to see that. Please download the new beta.

**Comment by fitdev** [comment]
Thank you for a quick fix! This new one seems to be working very well!

**Comment by wmjordan** [comment]
A new beta version of Codist 7.8 was just uploaded.
It may be the last beta for version 7.8.
If nothing goes bad, the version will be released in a few days.
